Document AssetBundle for Flutter (#4914)
getsentry/sentry-docs: src/platforms/flutter/common/performance/instrumentation/automatic-instrumentation.mdx

```

Slow and frozen frames are Mobile Vitals, which you can learn about in the [full documentation](/product/performance/mobile-vitals).

### AssetBundle Instrumentation

The [AssetBundle](https://api.flutter.dev/flutter/services/AssetBundle-class.html) instrumentation provides insight into how long your app takes to load its assets, such as files. It can be enabled by wrapping the `runApp` method with a `DefaultAssetBundle` and `SentryAssetBundle`.

```dart
import 'package:flutter/widgets.dart';
import 'package:sentry_flutter/sentry_flutter.dart';

Future<void> main() async {
  await SentryFlutter.init(
    (options) => options.dsn = '___PUBLIC_DSN___',
    appRunner: () => runApp(
      DefaultAssetBundle(
        bundle: SentryAssetBundle(),
        child: MyApp(),
      ),
    ),
  );
}
```

> The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `load`, `loadString`, and `loadStructuredData` call. The SDK sets the span `operation` to `file.read`.

Author: ```suggestion
The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `load` and `loadString` call. The SDK sets the span `operation` to `file.read`.
```

```

Slow and frozen frames are Mobile Vitals, which you can learn about in the [full documentation](/product/performance/mobile-vitals).

### AssetBundle Instrumentation

> The [AssetBundle](https://api.flutter.dev/flutter/services/AssetBundle-class.html) Instrumentation provides insight into how long your App takes to load its assets, e.g. files.

Reviewer: ```suggestion
The [AssetBundle](https://api.flutter.dev/flutter/services/AssetBundle-class.html) instrumentation provides insight into how long your app takes to load its assets, such as files. It can be enabled by wrapping the `runApp` method with a `DefaultAssetBundle` and `SentryAssetBundle`.
```

> 
> The `AssetBundle` instrumentation can be enabled by wrapping up the `runApp` method with a `DefaultAssetBundle` and `SentryAssetBundle`.
> 

Reviewer: ```suggestion
```

```dart
import 'package:flutter/widgets.dart';
import 'package:sentry_flutter/sentry_flutter.dart';

Future<void> main() async {
  await SentryFlutter.init(
    (options) => options.dsn = '___PUBLIC_DSN___',
    appRunner: () => runApp(
      DefaultAssetBundle(
        bundle: SentryAssetBundle(),
        child: MyApp(),
      ),
    ),
  );
}
```

> The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `load`, `loadString` and `loadStructuredData` call, The SDK sets the span `operation` to `file.read`.

Reviewer: ```suggestion
The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `load`, `loadString`, and `loadStructuredData` call. The SDK sets the span `operation` to `file.read`.
```

You can also enable the instrumentation for the `loadStructuredData` method by providing the `enableStructuredDataTracing` flag.

```dart
SentryAssetBundle(enableStructuredDataTracing: true)
```

> The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `loadStructuredData` call, The SDK sets the span `operation` to `serialize`.

Reviewer: ```suggestion
The `SentryAssetBundle` instrumentation starts a span out of the active transaction bound to the scope for each `loadStructuredData` call. The SDK sets the span `operation` to `serialize`.
```

<Note>

> The `enableStructuredDataTracing` Feature is available in Beta are still in-progress and may have bugs. We recognize the irony.

Reviewer: ```suggestion
The `enableStructuredDataTracing` feature is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
```
